examples/hubert/dataset/hubert_dataset.py

```diff
@@ -34,65 +33,77 @@ class BucketizeSampler(BatchSampler):
 
     def __init__(
         self,
-        data_source: Dataset,
+        lengths: List[int],
         num_buckets: int,
+        min_len: int = 0,
+        max_len: Optional[int] = None,
         max_token_count: Optional[int] = None,
         batch_size: Optional[int] = None,
+        shuffle: bool = True,
     ) -> None:
+        if max_len is None:
+            max_len = max(lengths)
+
+        if not (0 <= min_len <= max_len):
+            raise AssertionError("``min_len`` should be non-negative and smaller than ``max_len``")
         if max_token_count is not None and batch_size is not None:
             raise AssertionError("The ``max_token_count`` and ``batch_size`` can't be both set.")
-        self.data_source = data_source
+        # Filter out examples which are outside the bounds of [min_len, max_len]
+        self.lengths = [length for length in lengths if min_len <= length <= max_len]
+        if len(self.lengths) == 0:
+            raise AssertionError("``lengths`` cannot be empty after filtering.")
```

Comment:
``lengths`` needs to be sorted and a new indices list is used for batching. Although bucketization already gathered the samples with similar lengths, sorting the lengths can help reduce the gap when transferring to a new bucket.

For example:
batch_size = 16
batches = [0:8] is from bucket 0
bucket[8:16] is from bucket 100
The length different will be large.